Task1/Project/Shapes/Models/Classes/ShapeBase.cs

```diff
@@ -41,6 +61,12 @@ public CoordinateQuarters GetQuarter
                 else return middlePoint.Y > 0 ? CoordinateQuarters.Second : CoordinateQuarters.Third;
             }
         }
+        // CONSTRUCTOR
+        static ShapeBase()
+        {
+            dictionaryOfId = new Dictionary<string, Deserialization>();
+            idOfTheShape = (type) => type.Name;

```

Comment:
you should also registrate all your known classes here

Reply:
Done.